src/lib/io/sol-iio.c

```diff
@@ -1052,12 +1051,14 @@ sol_iio_read_channel_value(struct sol_iio_channel *channel, double *value)
     int len;
     int64_t raw_value;
     char path[PATH_MAX];
-    struct sol_iio_device *device = channel->device;
+    struct sol_iio_device *device;
     bool r;
 
     SOL_NULL_CHECK(channel, false);
     SOL_NULL_CHECK(value, false);
 
+    device = channel->device;
```

Comment:
@guchaojie Are you sure that `device` was `NULL`? This seems unlikely, as to add a channel one needs a device. I think that the bug here was `channel` being `NULL` (as no device was created), and trying to dereference it. If there's a channel with device NULL, we have bigger problems, and `SOL_NULL_CHECK` would just hide it.